### PR TITLE
React: Convert three more components

### PR DESCRIPTION
**Pull request by islemaster**
More cleanup.  Adding test coverage as I go.  These were fairly simple ones.

### CODE REVIEW

File: apps/src/templates/courseOverview/CourseOverview.js [hunk 1]
```diff
@@ -1,4 +1,5 @@
 import React, { Component, PropTypes } from 'react';
+import $ from 'jquery';
```
Comment (islemaster) [inline apps/src/templates/courseOverview/CourseOverview.js:2]: Caught this missing dependency while trying to run a subset of tests.

File: apps/src/templates/instructions/InstructionsWithWorkspace.jsx [hunk 2]
```diff
@@ -2,7 +2,7 @@ import $ from 'jquery';
 import React, {PropTypes} from 'react';
 import {connect} from 'react-redux';
 var CodeWorkspaceContainer = require('../CodeWorkspaceContainer');
-var TopInstructions = require('./TopInstructions');
+import TopInstructions from './TopInstructions';
```
Comment (islemaster) [inline apps/src/templates/instructions/InstructionsWithWorkspace.jsx:5]: Fixes a subtle problem - by adding a non-default export to the TopInstructions component for testing, the value loaded by a non-ES6 `require()` pattern changed from:

```
var TopInstructions = require('TopInstructions');
// Set to function (wrapped TopInstructions)
```

to

```
var TopInstructions = require('TopInstructions');
// Set to object {
//   default: function (wrapped TopInstructions),
//   UnconnectedTopInstructions: function UnconnectedTopInstructions
// }
```

Using an ES6 import resolves the problem by explicitly loading the default export, which was the original behavior.

File: apps/src/templates/instructions/InlineAudio.jsx [hunk 3]
```diff
@@ -159,27 +157,27 @@ const InlineAudio = React.createClass({
     const contentPath = `${hash}/${encodeURIComponent(message)}.mp3`;
 
     return `${TTS_URL}/${voicePath}/${contentPath}`;
-  },
+  }
 
-  toggleAudio: function () {
+  toggleAudio = () => {
```
Comment (balderdash) [inline apps/src/templates/instructions/InlineAudio.jsx:162]: Why do the toggle methods need to be arrow functions?

Comment (islemaster) [reply apps/src/templates/instructions/InlineAudio.jsx:162]: One of the changes when you switch from `createClass` to ES6 is that React no longer autobinds instance methods to the component instance, so if they get passed around as callbacks you have to bind them some other way.

`toggleAudio` is passed as a click callback here:

https://github.com/code-dot-org/code-dot-org/blob/c44242b3e038d6f31812f4440ade3f1b40b44863/apps/src/templates/instructions/InlineAudio.jsx#L201

Class property syntax and arrow functions is a nice way to autobind instance methods.

**Pull request merged**